### ISSUE TITLE
Crash on macos when clicking on the torrent 4.0.0-beta.2

### ISSUE THREAD

**Issue by Baccount** [transmission/transmission#4511]
### What is the issue?

The app crashed when I double clicked on a active torrent. Full crash log attached
```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
abort() called


Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1ade921b0 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x1adec8cec pthread_kill + 288
2   libsystem_c.dylib             	       0x1ade022c8 abort + 180
3   libsystem_malloc.dylib        	       0x1add20b1c malloc_vreport + 908
4   libsystem_malloc.dylib        	       0x1add37404 malloc_zone_error + 100
5   libsystem_malloc.dylib        	       0x1add2f204 nanov2_guard_corruption_detected + 44
6   libsystem_malloc.dylib        	       0x1add2e3fc nanov2_allocate_outlined + 404
7   CoreGraphics                  	       0x1b3005acc CGClipStackCreateMutableCopy + 40
8   CoreGraphics                  	       0x1b3005a88 maybeCopyClipState + 40
9   CoreGraphics                  	       0x1b300595c CGGStateClipToRect + 56
10  CoreGraphics                  	       0x1b30058c8 CGContextClipToRect + 176
11  AppKit                        	       0x1b1342358 -[NSImageCell drawInteriorWithFrame:inView:] + 268
12  AppKit                        	       0x1b166f0a4 __22-[NSControl drawRect:]_block_invoke + 96
13  AppKit                        	       0x1b1341e08 -[NSControl drawRect:] + 152
14  AppKit                        	       0x1b1341d58 -[NSImageView drawRect:] + 172
15  Transmission                  	       0x104adf2e8 0x104a8c000 + 340712
16  AppKit                        	       0x1b1341c38 _NSViewDrawRect + 124
17  AppKit                        	       0x1b19f4988 -[NSView _recursive:displayRectIgnoringOpacity:inContext:stopAtLayerBackedViews:] + 1088
18  AppKit                        	       0x1b13414ac -[NSView(NSLayerKitGlue) _drawViewBackingLayer:inContext:drawingHandler:] + 556
19  QuartzCore                    	       0x1b5653b2c CABackingStoreUpdate_ + 224
20  QuartzCore                    	       0x1b56afaa0 invocation function for block in CA::Layer::display_() + 64
21  QuartzCore                    	       0x1b5652d04 -[CALayer _display] + 1644
22  AppKit                        	       0x1b13411b4 -[_NSBackingLayer display] + 352
23  AppKit                        	       0x1b12d2cb4 -[_NSViewBackingLayer display] + 532
24  QuartzCore                    	       0x1b5651e88 CA::Layer::display_if_needed(CA::Transaction*) + 788
25  QuartzCore                    	       0x1b57b62d8 CA::Context::commit_transaction(CA::Transaction*, double, double*) + 456
26  QuartzCore                    	       0x1b5634404 CA::Transaction::commit() + 652
27  AppKit                        	       0x1b1350d58 __62+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayLink]_block_invoke + 272
28  AppKit                        	       0x1b1a2ece0 ___NSRunLoopObserverCreateWithHandler_block_invoke + 64
29  CoreFoundation                	       0x1adfa7dc8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 36
30  CoreFoundation                	       0x1adfa7cb4 __CFRunLoopDoObservers + 532
31  CoreFoundation                	       0x1adfa72ec __CFRunLoopRun + 784
32  CoreFoundation                	       0x1adfa6888 CFRunLoopRunSpecific + 612
33  HIToolbox                     	       0x1b767bfa0 RunCurrentEventLoopInMode + 292
34  HIToolbox                     	       0x1b767bc30 ReceiveNextEventCommon + 236
35  HIToolbox                     	       0x1b767bb2c _BlockUntilNextEventMatchingListInModeWithFilter + 72
36  AppKit                        	       0x1b1228424 _DPSNextEvent + 632
37  AppKit                        	       0x1b12275b4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 728
38  AppKit                        	       0x1b121b9e4 -[NSApplication run] + 464
39  AppKit                        	       0x1b11f2e28 NSApplicationMain + 880
40  dyld                          	       0x1adb9fe50 start + 2544

Thread 1:
0   libsystem_kernel.dylib        	       0x1ade8fe18 kevent + 8
1   Transmission                  	       0x104beb488 0x104a8c000 + 1438856
2   Transmission                  	       0x104be05b8 0x104a8c000 + 1394104
3   Transmission                  	       0x104b5b114 tr_session_thread_impl::sessionThreadFunc(event_base*) + 80
4   Transmission                  	       0x104b5b4e8 0x104a8c000 + 849128
5   libsystem_pthread.dylib       	       0x1adec906c _pthread_start + 148
6   libsystem_pthread.dylib       	       0x1adec3e2c thread_start + 8

Thread 2:
0   libsystem_kernel.dylib        	       0x1ade8d360 __semwait_signal + 8
1   libsystem_c.dylib             	       0x1add975b0 nanosleep + 220
2   Transmission                  	       0x104b19f3c 0x104a8c000 + 581436
3   Transmission                  	       0x104b8e198 tr_web::Impl::curlThreadFunc() + 400
4   Transmission                  	       0x104b8e474 0x104a8c000 + 1057908
5   libsystem_pthread.dylib       	       0x1adec906c _pthread_start + 148
6   libsystem_pthread.dylib       	       0x1adec3e2c thread_start + 8

Thread 3:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	       0x1ade89d70 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x1ade9b8a4 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x1ade925c4 mach_msg_overwrite + 540
3   libsystem_kernel.dylib        	       0x1ade8a0ec mach_msg + 24
4   CoreFoundation                	       0x1adfa8bc0 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x1adfa74ac __CFRunLoopRun + 1232
6   CoreFoundation                	       0x1adfa6888 CFRunLoopRunSpecific + 612
7   AppKit                        	       0x1b1352410 _NSEventThread + 172
8   libsystem_pthread.dylib       	       0x1adec906c _pthread_start + 148
9   libsystem_pthread.dylib       	       0x1adec3e2c thread_start + 8

Thread 4:
0   libsystem_pthread.dylib       	       0x1adec3e18 start_wqthread + 0

Thread 5:
0   libsystem_pthread.dylib       	       0x1adec3e18 start_wqthread + 0

Thread 6:
0   libsystem_pthread.dylib       	       0x1adec3e18 start_wqthread + 0


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x0000000000000000   x5: 0x0000000000000000   x6: 0x0000000000000001   x7: 0x0000000106ec8028
    x8: 0xbcab4988d03c8376   x9: 0xbcab498ad91e79f6  x10: 0xcccccccccccccccd  x11: 0x000000000000000a
   x12: 0x0000000000000000  x13: 0x0000000000000031  x14: 0x00000000000007fb  x15: 0x00000000a1e30820
   x16: 0x0000000000000148  x17: 0x000000020e0707f0  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x000000020922fa80  x21: 0x0000000000000103  x22: 0x000000020922fb60  x23: 0x000000016b36fd10
   x24: 0x0000000000000000  x25: 0x0000000000000000  x26: 0x000000016b373c77  x27: 0x000000020922fa80
   x28: 0x0000000206dcf000   fp: 0x000000016b36f660   lr: 0x00000001adec8cec
    sp: 0x000000016b36f640   pc: 0x00000001ade921b0 cpsr: 0x40001000
   far: 0x0000000106ed0000  esr: 0x56000080  Address size fault
```
[Transmission-2023-01-01-150132.ips.zip](https://github.com/transmission/transmission/files/10329619/Transmission-2023-01-01-150132.ips.zip)

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.0-beta.2 (bceb368f1b)

**Comment by ckerr** [comment]
@Baccount is this reproducible; and if so, could you give it another try with the latest nightly build @ https://build.transmissionbt.com/job/trunk-mac/ ?

**Comment by ckerr** [comment]
@Coeur @nevack any idea?

**Comment by Coeur** [comment]
Crash would be here:
https://github.com/transmission/transmission/blob/1e521b903030d4f1acc95615a24d37078126e88e/macosx/Controller.mm#L2239

But that seems perfectly normal API call.
Seems more like an Apple bug to me.

**Comment by Baccount** [comment]
@Coeur macOS 13.1 (22C65)

**Comment by Coeur** [comment]
While it could be an Apple bug, I'm not able to reproduce it.
From the code, it happened after a double click.
- Do you remember which tab was opened on the Inspector window (cmd+i)?
- Do you remember how many torrents were selected when you double-clicked, and where did you double click?

**Comment by Baccount** [comment]
1. I think it was the activity panel
2. Only one torrent was downloading and selected
@Coeur

**Comment by Coeur** [comment]
Sorry, I have no more clues since I can't reproduce it and the crash is in Apple API.
You could try to fill a report to Apple with Feedback Assistant.

**Comment by Coeur** [comment]
I reproduced once (and only once) this crash on `[NSWindow _doOrderWindow:]` and Apple said it's a possible heap memory corruption and it can come from anywhere. Then I never reproduced again after updating to macOS 13.2 beta 2 (22D5038i).

So, well, I still suspect an Apple bug, and invite you to send your reports to them via Feedback Assistant until they figure out something.

**Comment by KevinColemanInc** [comment]
I am experiencing this issue on my `Apple M1 Max` `macOS 13.2.1 (22D68)`. My machine has crashed twice now while downloading a magnetic link torrent [see the link here](https://news.ycombinator.com/item?id=35008347).

Interesting things:
- The torrent is large (~256GB).
- I am partially downloading a subset of the files.
- Downloading from ~20 peers
- Upload enabled
- wifi
- ~20 MB/s down.
- crashed downloading the same torrent
- changed wifi (but ran for ~20 min before crashing again)
- was not interacting with UI during crash (AFAIK)

```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x188db2868 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x188de9cec pthread_kill + 288
2   libsystem_c.dylib             	       0x188d222c8 abort + 180
3   libsystem_malloc.dylib        	       0x188c409f8 malloc_vreport + 908
4   libsystem_malloc.dylib        	       0x188c57404 malloc_zone_error + 100
5   libsystem_malloc.dylib        	       0x188c4ac7c free_list_checksum_botch + 40
6   libsystem_malloc.dylib        	       0x188c39424 small_free_list_remove_ptr_no_clear + 980
7   libsystem_malloc.dylib        	       0x188c346cc small_malloc_from_free_list + 392
8   libsystem_malloc.dylib        	       0x188c33eac small_malloc_should_clear + 220
9   libsystem_malloc.dylib        	       0x188c33cd0 szone_malloc_should_clear + 128
10  libsystem_trace.dylib         	       0x188b955d4 _os_log_fmt_flatten_object + 360
11  libsystem_trace.dylib         	       0x188b93340 _os_log_impl_flatten_and_send + 1864
12  libsystem_trace.dylib         	       0x188ba36e8 __os_signpost_emit_impl + 212
13  libsystem_trace.dylib         	       0x188b95c8c _os_signpost_emit_with_name_impl + 40
14  IconServices                  	       0x1951b57bc -[ISIcon _beginPrepareISIconSignpost:funcName:] + 512
15  IconServices                  	       0x1951b600c -[ISIcon prepareImagesForImageDescriptors:] + 56
16  AppKit                        	       0x18c3056b8 NSISIconImageRepGetCGImage + 220
17  AppKit                        	       0x18c265de8 __74-[NSImageRep drawInRect:fromRect:operation:fraction:respectFlipped:hints:]_block_invoke + 636
18  AppKit                        	       0x18c265a24 -[NSImageRep drawInRect:fromRect:operation:fraction:respectFlipped:hints:] + 632
19  AppKit                        	       0x18c6691dc __71-[NSImage drawInRect:fromRect:operation:fraction:respectFlipped:hints:]_block_invoke.1386 + 696
20  AppKit                        	       0x18c2486e8 -[NSImage _usingBestRepresentationForRect:context:hints:body:] + 148
21  AppKit                        	       0x18c265714 -[NSImage drawInRect:fromRect:operation:fraction:respectFlipped:hints:] + 1372
22  Transmission                  	       0x104cfdf5c 0x104cb4000 + 302940
23  AppKit                        	       0x18c37b9f4 -[NSTableView _drawContentsAtRow:column:withCellFrame:] + 232
24  AppKit                        	       0x18c405e9c -[NSOutlineView _drawContentsAtRow:column:withCellFrame:] + 64
25  AppKit                        	       0x18c37b498 -[NSTableView drawRow:clipRect:] + 824
26  Transmission                  	       0x104ce0dc0 0x104cb4000 + 183744
27  AppKit                        	       0x18c37ae80 -[NSTableView drawRowIndexes:clipRect:] + 464
28  AppKit                        	       0x18c4047e4 -[NSOutlineView drawRowIndexes:clipRect:] + 120
29  AppKit                        	       0x18c3171b4 -[NSTableView drawRect:] + 1172
30  AppKit                        	       0x18c2679c4 _NSViewDrawRect + 124
31  AppKit                        	       0x18c91a5c8 -[NSView _recursive:displayRectIgnoringOpacity:inContext:stopAtLayerBackedViews:] + 1088
32  AppKit                        	       0x18c267238 -[NSView(NSLayerKitGlue) _drawViewBackingLayer:inContext:drawingHandler:] + 556
33  QuartzCore                    	       0x19057df40 CABackingStoreUpdate_ + 224
34  QuartzCore                    	       0x1905d9f6c invocation function for block in CA::Layer::display_() + 64
35  QuartzCore                    	       0x19057d118 -[CALayer _display] + 1644
36  AppKit                        	       0x18c266f40 -[_NSBackingLayer display] + 352
37  AppKit                        	       0x18c1f8968 -[_NSViewBackingLayer display] + 532
38  QuartzCore                    	       0x19057c29c CA::Layer::display_if_needed(CA::Transaction*) + 788
39  QuartzCore                    	       0x1906e0904 CA::Context::commit_transaction(CA::Transaction*, double, double*) + 456
40  QuartzCore                    	       0x19055e818 CA::Transaction::commit() + 652
41  AppKit                        	       0x18c276ae4 __62+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayLink]_block_invoke + 272
42  AppKit                        	       0x18c954a28 ___NSRunLoopObserverCreateWithHandler_block_invoke + 64
43  CoreFoundation                	       0x188ec8db8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 36
44  CoreFoundation                	       0x188ec8ca4 __CFRunLoopDoObservers + 532
45  CoreFoundation                	       0x188ec82dc __CFRunLoopRun + 784
46  CoreFoundation                	       0x188ec7878 CFRunLoopRunSpecific + 612
47  HIToolbox                     	       0x1925a7fa0 RunCurrentEventLoopInMode + 292
48  HIToolbox                     	       0x1925a7de4 ReceiveNextEventCommon + 672
49  HIToolbox                     	       0x1925a7b2c _BlockUntilNextEventMatchingListInModeWithFilter + 72
50  AppKit                        	       0x18c14d84c _DPSNextEvent + 632
51  AppKit                        	       0x18c14c9dc -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 728
52  AppKit                        	       0x18c140e0c -[NSApplication run] + 464
53  AppKit                        	       0x18c118250 NSApplicationMain + 880
54  dyld                          	       0x188abfe50 start + 2544
```

**Comment by ckerr** [comment]
Closing this ticket as stale due to the age of the crash and the lack of reproducible info.

If this is still happening in >= 4.1.0 please open a new ticket with an updated crash report. Thank yoU!
